File: kiosk-backend/src/main/kotlin/com/kiosk/domain/model/Store.kt
package com.kiosk.domain.model

import java.time.LocalTime

data class Store(
    val id: Long = 1,
    val name: String,
    val openTime: LocalTime = LocalTime.of(9, 0),
    val closeTime: LocalTime = LocalTime.of(22, 0),
    val isOpen: Boolean = true,
    val hideAdminButton: Boolean = false
)

File: kiosk-app/lib/models/store.dart
class Store {
  final int id;
  final String name;
  final String openTime;
  final String closeTime;
  final bool isOpen;
  final bool hideAdminButton;

  Store({
    required this.id,
    required this.name,
    required this.openTime,
    required this.closeTime,
    required this.isOpen,
    this.hideAdminButton = false,
  });

  factory Store.fromJson(Map<String, dynamic> json) => Store(
        id: json['id'] as int,
        name: json['name'] as String,
        openTime: json['openTime'] as String,
        closeTime: json['closeTime'] as String,
        isOpen: json['isOpen'] as bool,
        hideAdminButton: json['hideAdminButton'] as bool? ?? false,
      );
}
